core/trino-main/src/main/java/io/trino/server/TaskResource.java

```diff
@@ -138,11 +142,17 @@ public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo)
     @Produces(MediaType.APPLICATION_JSON)
     public void createOrUpdateTask(
             @PathParam("taskId") TaskId taskId,
+            @QueryParam("nodeId") String requestNodeId,
             TaskUpdateRequest taskUpdateRequest,
             @Context UriInfo uriInfo,
             @Suspended AsyncResponse asyncResponse)
     {
         requireNonNull(taskUpdateRequest, "taskUpdateRequest is null");
+        if (!nodeIdentifier.equals(requestNodeId)) {
+            asyncResponse.resume(Response.status(Status.BAD_REQUEST)
```

Comment:
you should return here as this will continue to execute but the response is already resumed

Reply:
good catch